### PR TITLE
Add unit test folder

### PR DESCRIPTION
**Pull request by davaya**
Include an empty test folder in the tutorial package layout.  Addresses Issue #619.

### CODE REVIEW

File: source/tutorials/packaging-projects.rst [hunk 1]
```diff
@@ -44,11 +44,17 @@ them in the following steps.
...
         __init__.py
+      test/
       setup.py
       LICENSE
       README.md
 
 
+Creating a test folder
+----------------------
+:file:`test/` is a placeholder for unit test files. Leave it empty for now.
```
Comment (pradyunsg) [inline source/tutorials/packaging-projects.rst]:
```suggestion

:file:`tests/` is a placeholder for unit test files. Leave it empty for now.
```

File: source/tutorials/packaging-projects.rst [hunk 2]
```diff
@@ -44,11 +44,17 @@ them in the following steps.
     packaging_tutorial/
       example_pkg/
         __init__.py
+      test/
```
Comment (pradyunsg) [inline source/tutorials/packaging-projects.rst]:
```suggestion
      tests/
```

**Comment by davaya** [comment]
> This will instantly turn into a massive bikeshedding exercise ...

I envisioned https://www.cbsnews.com/news/brooklyn-bike-race-red-hook-criterium-sees-large-pileup-at-start/, but discovered Parkinson's law of triviality.  Thanks for a new word for the day.

**Comment by pfmoore** [comment]
> Thanks for a new word for the day.

Cool - you're one of today's [lucky 10000](https://xkcd.com/1053/) 🙂

**Comment by webknjaz** [comment]
Should this mention that there's also a legitimate reason to put `tests/` under the package for distributing as an sdist?

**Comment by pfmoore** [comment]
IMO no. If we're not willing to agree on a placeholder location in the very initial example, then so be it, we should do nothing and keep the status quo. But @davaya made the point in https://github.com/pypa/packaging.python.org/issues/619#issuecomment-587567723 that it would be good to encourage newcomers to include some sort of tests from the start.

**Comment by webknjaz** [comment]
@pfmoore FWIW I agree that most of the time `tests/` should be in the root folder. Just wanted to explore an opportunity to improve the wording maybe.

**Comment by pfmoore** [comment]
Do you have a suggestion for what to add? I'm open to consider specific wording, I just don't want to get into broad debates (and I don't know how I'd word such a comment myself).

File: source/tutorials/packaging-projects.rst [hunk 3]
```diff
@@ -44,11 +44,18 @@ them in the following steps.
...
+      tests/
       setup.py
       LICENSE
       README.md
 
 
+Creating a test folder
+----------------------
+
+:file:`tests/` is a placeholder for unit test files. Leave it empty for now.
```
Comment (webknjaz) [inline source/tutorials/packaging-projects.rst]:
```suggestion
:file:`tests/` is a placeholder for unit test files. Leave it empty for now.

  .. note::

     In *rare* cases, you may meet some projects that put the ``tests/`` folder under
     ``example_pkg/``. They probably use `pytest's pyargs`_. While there may be a
     legitimate reason to do so, you'll likely never need it. So for the sake of
     simplicity, this guide will stick with the most common case of a top-level
     ``tests/`` dir (in plural).

     .. _pytest's pyargs:
     https://docs.pytest.org/en/latest/goodpractices.html#tests-as-part-of-application-code
```

@pfmoore ^

Comment (webknjaz) [inline source/tutorials/packaging-projects.rst]: Nitpick: I'd probably remove "unit" from there. Because other types of tests exist and can be put there.

Comment (pfmoore) [reply source/tutorials/packaging-projects.rst]: That seems too wordy to me, in that it obscures the key point. Maybe just 

```
:file:`tests/` is a placeholder for unit test files (other test layouts are possible, but this is a good starting point). Leave it empty for now.
```

?

Comment (webknjaz) [reply source/tutorials/packaging-projects.rst]: Yeah, that sounds good. Should we keep the link?

```suggestion
:file:`tests/` is a placeholder for test files (`other test layouts`_ are possible, but this is a good starting point). Leave it empty for now.

.. _other test layouts:
   https://docs.pytest.org/en/latest/goodpractices.html#tests-as-part-of-application-code
```

@pfmoore ^

Comment (pfmoore) [reply source/tutorials/packaging-projects.rst]: I'd argue no. If we link to a discussion of that layout, we should also link to a discussion of the top-level `tests` layout, and once again we spiral into bikeshedding.

**Comment by ncoghlan** [comment]
I went ahead and merged the simple version (like @pf_moore, I think @davaya made a very solid point in https://github.com/pypa/packaging.python.org/issues/619#issuecomment-587567723).

We can continue discussion in #619 on how we can build on this to introduce the topics of:
* ensuring sdists include tests, not just the installed package
* ensuring the installed package works, not just the files in the source repo
* keeping built wheel files efficient by omitting unnecessary files (such as tests)
